modules/4337/certora/specs/SignatureLengthCheck.spec

```diff
@@ -0,0 +1,25 @@
+using Account as safeContract;
+
+methods {
+    function checkSignaturesLength(bytes, uint256) external returns(bool) envfree;
+    function combineBytes(bytes, bytes) external returns(bytes) envfree;
+
+    // Safe Contract functions
+    function safeContract.canonicalSignatureHash(bytes, uint256) external returns(bytes32) envfree;
+}
+
+// This rule verifies that if excess data is added to the dynamic part of the signature, then the signature check will fail.
```

Comment:
```suggestion
// This rule verifies that if excess data is added to the the signature, then the signature check will fail.
```

Reply:
The rule doesn't seem to be specific to contract signature.

Reply:
Updated, the initial intention to write like this was because the excess data can only be added to the dynamic part and not to the static part.